tf2onnx/tfonnx.py

```diff
@@ -1354,11 +1354,11 @@ def onehot_op(ctx, node, name, args):
     axis = node.get_attr("axis")
     # axis becomes axis for gather
     node.set_attr("axis", 0)
-    depth = node.inputs[1].get_tensor_value()[0]
-    on = node.inputs[2].get_tensor_value()[0]
-    off = node.inputs[3].get_tensor_value()[0]
-    dtype = node.inputs[2].get_tensor_type()
-    eye = np.eye(depth, dtype=dtype)
+    depth = node.inputs[1].get_tensor_value()
+    on_val = node.inputs[2].get_tensor_value(as_list=False)
+    on = on_val.tolist()
+    off = node.inputs[3].get_tensor_value()
```

Comment:
off is a list ?

Reply:
for a scalar, tolist() won't be a real list, it is still a single value. while tolist() changed the return value's type.